### PR TITLE
Update waymo dataset, evaluation metrics and related configs for 2.0

### PR DESCRIPTION
**Pull request by lianqing11**
## Motivation

Update waymo dataset, evaluation metrics and related configs for 2.0 with pointpillars and second

### CODE REVIEW

File: mmdet3d/datasets/waymo_dataset.py [hunk 1]
```diff
@@ -26,524 +23,199 @@ class WaymoDataset(KittiDataset):
...
             Defaults to False.
-        pcd_limit_range (list(float), optional): The range of point cloud used
-            to filter invalid predicted boxes.
+        pcd_limit_range (list, optional): The range of point cloud used to
+            filter invalid predicted boxes.
             Default: [-85, -85, -5, 85, 85, 5].
+        cam_sync_instances (bool, optional): If use the camera sync label
+            supported from waymo version 1.3.1.
+        load_interval (int, optional): load frame interval.
+        task (str, optional): task for 3D detection (lidar, mono3d, and etc).
```
Comment (Tai-Wang) [inline mmdet3d/datasets/waymo_dataset.py:51]: May list the currently supported options explicitly

File: tools/data_converter/update_infos_to_v2.py [hunk 2]
```diff
@@ -791,6 +795,222 @@ def update_lyft_infos(pkl_path, out_dir):
...
+
+    mmcv.dump(converted_data_info, out_path, 'pkl')
+
+
+def generate_waymo_camera_instances(ori_info_dict, cam_keys):
+
+    empty_multicamera_instances = get_empty_multicamera_instances(cam_keys)
+
+    for cam_idx, cam_key in enumerate(cam_keys):
+        # cam_info = info['cams'][cam]
```
Comment (Tai-Wang) [inline tools/data_converter/update_infos_to_v2.py:1002]: can remove

File: mmdet3d/evaluation/metrics/waymo_metric.py [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+                                xywhr2xyxyr)
+from .kitti_metric import KittiMetric
+
+
+@METRICS.register_module()
+class WaymoMetric(KittiMetric):
+    """Waymo evaluation metric.
+
+    Args:
+        data_root (str): Path of dataset root.
```
Comment (Tai-Wang) [inline mmdet3d/evaluation/metrics/waymo_metric.py:24]: Need to update the docstring accordingly

File: mmdet3d/evaluation/metrics/waymo_metric.py [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+        self.waymo_bin_file = waymo_bin_file
+        self.data_root = data_root
+        self.file_client_args = file_client_args
+        self.split = split
+        self.task = task
+        self.use_pred_sample_idx = use_pred_sample_idx
+        super().__init__(
+            ann_file=ann_file,
+            metric=metric,
+            # pred_box_type_3d=pred_box_type_3d,
```
Comment (Tai-Wang) [inline mmdet3d/evaluation/metrics/waymo_metric.py:80]: can remove

File: mmdet3d/datasets/waymo_dataset.py [hunk 5]
```diff
@@ -26,524 +23,201 @@ class WaymoDataset(KittiDataset):
     Args:
         data_root (str): Path of dataset root.
         ann_file (str): Path of annotation file.
-        split (str): Split of input data.
-        pts_prefix (str, optional): Prefix of points files.
-            Defaults to 'velodyne'.
+        data_prefix (list[dict], optional): data prefix for point cloud and
```
Comment (Tai-Wang) [inline mmdet3d/datasets/waymo_dataset.py:26]: No need for optional for new version of mmdet3d
Still need to show the default value

File: mmdet3d/evaluation/metrics/waymo_metric.py [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+        if self.pklfile_prefix is None:
+            eval_tmp_dir = tempfile.TemporaryDirectory()
+            pklfile_prefix = osp.join(eval_tmp_dir.name, 'results')
+        else:
+            eval_tmp_dir = None
+            pklfile_prefix = self.pklfile_prefix
+
+        # load annotations
+
+        waymo_bin_file = self.waymo_bin_file
```
Comment (Tai-Wang) [inline mmdet3d/evaluation/metrics/waymo_metric.py:117]: Why need this? Can remove?

File: mmdet3d/evaluation/metrics/waymo_metric.py [hunk 7]
```diff
@@ -0,0 +1,443 @@
...
+            waymo_results_save_dir,
+            waymo_results_final_path,
+            prefix,
+            file_client_args=self.file_client_args)
+        converter.convert()
+        waymo_save_tmp_dir.cleanup()
+        return result_files, waymo_save_tmp_dir
+
+    def merge_multi_view_boxes(self, box_dict_per_frame, cam0_info):
+        """Merge bounding boxes predicted from multi-view images."""
```
Comment (Tai-Wang) [inline mmdet3d/evaluation/metrics/waymo_metric.py:245]: Need docstring for other two variables

File: mmdet3d/evaluation/metrics/waymo_metric.py (hunk 6 above)
Comment (lianqing11) [reply mmdet3d/evaluation/metrics/waymo_metric.py]: remove this

**Pull request merged**